### PR TITLE
test: windows tarballs

### PR DESCRIPTION
**Pull request by mdonnalley**
### What does this PR do?

Run JIT tests for linux and windows

Depends on https://github.com/salesforcecli/plugin-release-management/pull/868

### What issues does this PR fix or reference?
@W-14096048@

### CODE REVIEW

File: .github/workflows/test.yml [hunk 1]
```diff
@@ -20,6 +20,39 @@ jobs:
...
+      - uses: actions/upload-artifact@v2
+        with:
+          name: manifest
+          path: oclif.manifest.json
+
+  jit-install:
+    needs: [manifest]
+    strategy:
+      matrix:
+        os: [ubuntu-latest, windows-latest]
```
Comment (iowillhoit) [inline .github/workflows/test.yml:42]: Wouldn't this cause the JIT installs to run twice for linux? They will have already ran in the `tarballs` job above ([source](https://github.com/salesforcecli/github-workflows/blob/d92532a5b5f76b77de2e080d2c51da8fa4a58859/.github/workflows/tarballs.yml#L50C10-L50C10))

File: package.json [hunk 2]
```diff
@@ -233,7 +233,7 @@
     "@oclif/plugin-command-snapshot": "^3.3.0",
     "@salesforce/dev-config": "^4.0.1",
     "@salesforce/dev-scripts": "^3.1.1",
-    "@salesforce/plugin-release-management": "^4.1.19",
+    "@salesforce/plugin-release-management": "^4.1.29-dev.0",
```
Comment (iowillhoit) [inline package.json:236]: Just adding a comment so we remember to update this.

File: .github/workflows/test.yml (hunk 1 above)
Comment (mdonnalley) [reply .github/workflows/test.yml:42]: @iowillhoit Yeah I wasn't sure what to do about that.

The existing test will run the smoke tests (which include JIT tests) on the tarballs produced by `oclif pack` whereas these new tests run the JIT tests on the local `bin/run`. I felt weird about removing the JIT tests from the smoke tests and I felt weird about only running these new tests only on windows

I'll happily defer to your judgement here though

Comment (iowillhoit) [reply .github/workflows/test.yml:42]: Ah that is a good distinction. I am fine leaving this testing both. Honestly linux will finish faster than windows anyway so it wont be adding any more time.

**Comment by iowillhoit** [comment]
You'll need to remove the `--cli sf` from the `package.json`
`"test:smoke-unix": "sf-release cli:tarballs:smoke --cli sf",`

**Review by iowillhoit — APPROVED** [review]

**Pull request merged**